docs/proposals/service-design.md

```diff
@@ -0,0 +1,85 @@
+---
+title: Service Design
+status: Pending
+authors:
+    - "@qizha"
+approvers:
+  - "@kevin-wangzefeng"
+  - "@CindyXing"
+creation-date: 2019-03-20
+last-updated: 2019-03-20
+---
+
+# Service support on edge
+
+## Abstract
+To support service capabilities on edge to support microservice communication cross cloud and edge.
+The service discovery, communication should be considered in this feature.
+Developers can define services on top of the deployments deployed on the cloud or edges in a unified manner without caring about their location. The service name is the only thing they need to know to do the communication. For example, the following address can be used for a REST  request:
+`http://{service_name}/resources`
+
+## Motivation
+The cloud native and microservice architecture is becoming more and more popular and the edge node is becoming more and more powerful. The user want to decompose their application into multiple microservices. Some of those microservices need to be deployed in edge node to get close to the data generated by devices. By simplify the development, the user need to use the same michanism to do the service discovery and communication with what they are doing on cloud.
+
+## Constraints and Assumptions
+1. No DNS services available for edge.
+2. The identification and authentication should be handled by microservices.
+3. When service communication between edge to edge, they may work offline. With the purpose to deliver a "expected result" to the user system, the service definition should be push down to the edge right after the definition operation. That is to say, the istio model may not be suitable for edge, in istio model, the service definition is pull from K8S master when the service is invoked.
+4. Only support HTTP communication in the first step
+5. All edge nodes are in the same sub-network and they have the network reachability.
+
+## Use cases
+<img src="../images/proposals/service-design.png">
+
+### Register the service located on cloud
+* CS0.1: User create service(type ClusterIP) in K8S API Server
+* CS0.2: EC get the service definition and create router rules with the edge servicebus as source and service clusterIP as target.
+* CS0.3: Send the service definition to edge
+
+### Register the service located on edge
+* CS1: User create service(headless service without ClusterIP) in K8S API Server
```

Comment:
Any reason we don't support ClusterIP on edge and I am wondering how headless service work without DNS servers.

Reply:
1. I want to keep the edge part simple especially for the network. And it is a websocket connection between cloud and edge, considering some services are located on cloud and some on the edge. A overlay network will be required to build the overall clusterIP system. It will be too complicate. 
2. This way we can leverage the service mesh ideas to build some more flow control capabilities on top of it
For the headless service, we just need a service name with a set of selected POD. Application can visit another one with the service name just like what istio provided.

Reply:
Okay, thanks for clarification.